src/Composer/Repository/ArtifactRepository.php

```diff
@@ -88,7 +88,12 @@ private function scanDirectory($path)
 
     private function getComposerInformation(\SplFileInfo $file)
     {
-        $json = Zip::getComposerJson($file->getPathname());
+        $json = null;
+        try {
+            $json = Zip::getComposerJson($file->getPathname());
+        } catch (\Exception $exception) {
+            $this->io->write($exception->getMessage(), false, IOInterface::VERBOSE);
```

Comment:
If you have an artifact repo with a 100 archive and one of them fails to load it will just output the exception message which is "No composer.json found either at the top level or within the topmost directory" for example. That does not seem very useful to the user if they don't know which archive it was in. I'd suggest adding more context before the exception message here: `'Failed loading package '.$file->getPathname().': '.$exception->getMessage()` or something like that.